src/main/java/com/hotels/bdp/circustrain/bigquery/partition/PartitionQueryFactory.java

```diff
@@ -31,14 +31,14 @@
   public String newInstance(Table hiveTable, String partitionBy, String partitionFilter) {
     if (isNotBlank(partitionBy) && isNotBlank(partitionFilter)) {
       String query = String
-          .format("select distinct(%s) from %s.%s where %s group by %s order by %s", partitionBy, hiveTable.getDbName(),
-              hiveTable.getTableName(), partitionFilter, partitionBy, partitionBy);
+          .format("select distinct(%s) from %s.%s where %s order by %s", partitionBy, hiveTable.getDbName(),
```

Comment:
Why are the `group by` statements no longer needed?

Reply:
So before it was `select x from db.table where <condition> group by x order by x` and replacing it with `select distinct(x) from db.table where <condition> order by x` means the same thing, but is more specific to what the query is trying to get: a distinct list of all the values in the column.

I was looking into the difference between `distinct` and `group by` and found this: https://stackoverflow.com/questions/164319/is-there-any-difference-between-group-by-and-distinct.

Reply:
OK right, yes if you are selecting just X then it makes no sense to group by it as you will only ever have one group - X.